cloudsql-ha-dr/readme-Update (#48)
GoogleCloudPlatform/click-to-deploy-solutions: cloudsql-ha-dr/README.md

[![banner](../banner.png)](https://cloud.google.com/?utm_source=github&utm_medium=referral&utm_campaign=GCP&utm_content=packages_repository_banner)

# Host a Highly Available SQL database on Google Cloud

## Introduction

> This architecture uses click-to-deploy to demonstrate how to deploy a CloudSQL instance with high-availability and cross-region replica.

Reviewer: ```suggestion
This architecture uses click-to-deploy to demonstrate how to deploy a Cloud SQL instance with high-availability and cross-region replica.
```

Resources created:
- VPC
- CloudSQL for Postgres instance with [high-availability](https://cloud.google.com/sql/docs/postgres/high-availability)
- CloudSQL [cross-region read replica](https://cloud.google.com/sql/docs/postgres/intro-to-cloud-sql-disaster-recovery) instance
With data being a vital asset for organizations, the high availability and disaster recovery capabilities offered by Cloud SQL allow businesses to meet stringent data availability and regulatory requirements.

By implementing this solution, organizations gain peace of mind knowing that their critical databases are protected and accessible, even during unforeseen events or outages. This solution ensures business continuity by minimizing downtime, providing robust data protection through automated backups, and enabling seamless failover and disaster recovery processes.

:clock1: Estimated deployment time: 16 min 30 sec
This solution is suitable for a wide range of applications and industries, including e-commerce platforms, financial systems, customer relationship management (CRM) tools, and more.

## Use cases

These are some examples of the use cases you can build on top of this architecture:

> :heavy_dollar_sign: Estimated solution cost: [USD 706.44 per 1 month](https://cloud.google.com/products/calculator/#id=358b4a68-4f82-4e88-84f4-1da99a05548b)

Reviewer: I see you removed the estimation here but did not add it to the costs section. Could you please take a look?

Author: I actually moved this to the cost section under the architecture, the text changed but the link to the calculator is the same.

<p align="center"><img src="architecture.png"></p>

The main components that we would be setting up are (to learn more about these products, click on the hyperlinks).

* [VPC](https://cloud.google.com/vpc) : Global virtual network that spans all regions. Single VPC for an entire organization, isolated within projects. Increase IP space with no downtime.

> * CloudSQL for Postgres instance with [high-availability](https://cloud.google.com/sql/docs/postgres/high-availability) : The purpose of an HA configuration is to reduce downtime when a zone or instance becomes unavailable. 

Reviewer: ```suggestion
* Cloud SQL for Postgres instance with [high-availability](https://cloud.google.com/sql/docs/postgres/high-availability) : The purpose of an HA configuration is to reduce downtime when a zone or instance becomes unavailable. 
```

> * CloudSQL [cross-region read replica](https://cloud.google.com/sql/docs/postgres/intro-to-cloud-sql-disaster-recovery) instance : database disaster recovery (DR) is about providing continuity of processing, specifically when a region fails or becomes unavailable.

Reviewer: ```suggestion
* Cloud SQL [cross-region read replica](https://cloud.google.com/sql/docs/postgres/intro-to-cloud-sql-disaster-recovery) instance : database disaster recovery (DR) is about providing continuity of processing, specifically when a region fails or becomes unavailable.
```
